web/src/realtime-info-signal/middleware.js

```diff
@@ -0,0 +1,73 @@
+import {
+  OPEN_EXPANDED_VIEW,
+  CLOSE_EXPANDED_VIEW,
+} from '../expanded-view/actions'
+import {
+  START_TITLE_EDIT,
+  END_TITLE_EDIT,
+  START_DESCRIPTION_EDIT,
+  END_DESCRIPTION_EDIT,
+} from '../goal-editing/actions'
+import { sendRealtimeInfoSignal } from './actions'
+
+
+
+// capture open/close expanded view, start/end title/description edit, and enter/exit project actions
+// import action constants
+// isOneOfRealtimeInfoAffectingActions checks action type against
+// can first start with just open close expanded view, then change the title/description edits action logic (and state) then implement in middleware
+
+const isOneOfRealtimeInfoAffectingActions = (action) => {
+  const { type } = action
+  return type === OPEN_EXPANDED_VIEW
+      || type === CLOSE_EXPANDED_VIEW
+      || type === START_TITLE_EDIT
+      || type === END_TITLE_EDIT
+      || type === START_DESCRIPTION_EDIT
+      || type === END_DESCRIPTION_EDIT
+}
+// watch for actions that will affect the realtime info state values
+
+const realtimeInfoWatcher = store => {
+
+
+  // return the action handler middleware
+  return next => async action => {
+
+    const shouldSendRealtimeSignal = isOneOfRealtimeInfoAffectingActions(action)
+    if (shouldSendRealtimeSignal) {
+      console.log('logging middleware:', action.payload) 
+      let result = next(action)
+      let state = store.getState()
```

Comment:
as you've done here, this is good. 

it's important to call `next(action)` before calling `store.getState()` because otherwise it won't have integrated into the store, the main action that we are intercepting (such as START_TITLE_EDIT) and thus the `getRealtimeInfo(state)` below won't have the right state details. 

In effect, we are chaining this action on *after* the other one